instrumentation-genai/opentelemetry-instrumentation-openai-v2/src/opentelemetry/instrumentation/openai_v2/patch.py

```diff
@@ -149,6 +150,137 @@ async def traced_method(wrapped, instance, args, kwargs):
+                instance,
+                GenAIAttributes.GenAiOperationNameValues.EMBEDDINGS.value,
+            )
+        }
+
+        # Using a custom attribute "gen_ai.embeddings.dimensions". Will propose to semantic conventions.
+        if "dimensions" in kwargs and kwargs["dimensions"] is not None:
+            span_attributes["gen_ai.embeddings.dimensions"] = kwargs[
+                "dimensions"
+            ]
+
+        span_name = f"{span_attributes[GenAIAttributes.GEN_AI_OPERATION_NAME]} {span_attributes[GenAIAttributes.GEN_AI_REQUEST_MODEL]}"
+        with tracer.start_as_current_span(
+            name=span_name,
+            kind=SpanKind.CLIENT,
+            attributes=span_attributes,
+            end_on_exit=False,
+        ) as span:
+            # Store the input for later use in the response attributes
+            input_text = kwargs.get("input", "")
+
+            start = default_timer()
+            result = None
+            error_type = None
+            try:
+                result = wrapped(*args, **kwargs)
+
+                if span.is_recording():
+                    _set_embeddings_response_attributes(
+                        span, result, event_logger, capture_content, input_text
+                    )
+
+                span.end()
+                return result
+
+            except Exception as error:
+                error_type = type(error).__qualname__
+                handle_span_exception(span, error)
+                raise
+            finally:
+                duration = max((default_timer() - start), 0)
+                _record_metrics(
+                    instruments,
+                    duration,
+                    result,
+                    span_attributes,
+                    error_type,
+                    GenAIAttributes.GenAiOperationNameValues.EMBEDDINGS.value,
+                )
+
+    return traced_method
+
+
+def async_embeddings_create(
+    tracer: Tracer,
+    event_logger: EventLogger,
+    instruments: Instruments,
+    capture_content: bool,
+):
+    """Wrap the `create` method of the `AsyncEmbeddings` class to trace it."""
+
+    async def traced_method(wrapped, instance, args, kwargs):
+        span_attributes = {
+            **get_llm_request_attributes(
+                kwargs,
+                instance,
+                GenAIAttributes.GenAiOperationNameValues.EMBEDDINGS.value,
+            )
+        }
+
+        # Set embeddings dimensions if specified in the request
+        if "dimensions" in kwargs and kwargs["dimensions"] is not None:
+            span_attributes["gen_ai.embeddings.dimensions"] = kwargs[
+                "dimensions"
+            ]
+
+        span_name = f"{span_attributes[GenAIAttributes.GEN_AI_OPERATION_NAME]} {span_attributes[GenAIAttributes.GEN_AI_REQUEST_MODEL]}"
+        with tracer.start_as_current_span(
+            name=span_name,
+            kind=SpanKind.CLIENT,
+            attributes=span_attributes,
+            end_on_exit=False,
```

Comment:
why not close it on exit and remove line 268 (`span.end`) ? there is no streaming in this case